src/ApparentHorizons/Tags.hpp

```diff
@@ -12,7 +12,9 @@
 #include "Utilities/ForceInline.hpp"
 #include "Utilities/TMPL.hpp"
 
-class DataVector;
+template <typename T>
+class DataVectorImpl;
+using DataVector = DataVectorImpl<double>;
```

Comment:
We discussed using a strong typedef rather than a type alias so you don't need to do this everywhere. Then you can forward declare the strong type def. That is,

```
struct DataVector: public DataVectorImpl<double> {
  using DataVectorImpl<double>::DataVectorImpl;
  // Any other functions that should become visible
};
```

Reply:
It looks like there are a couple of things that need to be declared in the `struct DataVector` for this, but I think I'll be able to keep it pretty minimal. I'll revise with the strong typedef version.

Reply:
It appears to be true that if we use inheritance rather than the transparent typedef, we will also need to re-implement potentially templatable operators, limiting the usefulness of the templating structure (particularly e.g. `==` acting on arrays of DataVectors -- there are several such implementations in DataVector.hpp). It is my understanding that the compiler will not directly use the operators associated with the base class in this case (when the called arguments are `array<Derived, dim>`, strong typedef ensures that it will not template deduce to a declared  `array<Base<spec>,dim>`, even when Derived inherits from Base<spec>).
Does this change any perspectives on strong typedef vs `using` declaration?
short version: do we want annoying forward DataVector declares, or code duplication for operators on DataVectors of different types?

Reply:
Okay, I think I like using macros for these -- I'll put that in.